lib/logstash/filters/csv.rb

```diff
@@ -151,10 +151,14 @@ def filter(event)
   private
 
   # construct the correct Event field reference for given field_name, taking into account @target
-  # @param field_name [String] the bare field name without brakets
+  # @param field_name [String] the field name.
   # @return [String] fully qualified Event field reference also taking into account @target prefix
   def field_ref(field_name)
-    "#{@target}[#{field_name}]"
+    if field_name.start_with?("[")
```

Comment:
Indeed, the bug was this: If a user specified `columns => [ "foo", "[bar][baz]" ]`, then this second column (well, any column using nested field references, or brackets in general) would end up being transformed like this:

`<target>[<field>]` 

and in this case our field is `[bar][baz]` which results in `[[bar][baz]]` which is not what we want. 

The intent here is because 'target' could exist, such as in this config:

```
target => "foo"
columns => [ "one", "[two][bar]" ]
```

The user is telling us column 1 is named `[foo][one]` and column 2 is named `[foo][two][bar]`. Prior to this patch, the second column would be incorrectly called `[foo][[two][bar]]` with extra brackets around the `[two][bar]`